src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConstructorInitTests.cs

```diff
@@ -607,5 +737,488 @@ public void BadExpressionConstructor()
+        {
+            string source = @"
+#nullable enable
+
+struct S
+{
+    public int x;
+    public int y;
+}
+
+class C
+{
+    static S field1 = default;
+    static S field2 = default(S);
+    static S field3 = new S();
+}";
+            // note: we could make the synthesized constructor smarter and realize that
+            // nothing needs to be emitted for these initializers.
+            // but it doesn't serve any realistic scenarios at this time.
+            CompileAndVerify(source).VerifyIL("C..cctor()", @"
+{
+  // Code size        1 (0x1)
+  .maxstack  0
+  IL_0000:  ret
+}");
+        }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_07()
+        {
+            string source = @"
+#nullable enable
+class C
+{
+    static int x = 0;
+}";
+            CompileAndVerify(
+                source,
+                symbolValidator: validator,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All));
+
+            void validator(ModuleSymbol module)
+            {
+                var type = module.ContainingAssembly.GetTypeByMetadataName("C");
+                Assert.Null(type.GetMember(".cctor"));
+            }
+        }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_08()
+        {
+            string source = @"
+#nullable enable
+class C
+{
+    static int x = 1;
+}";
+            CompileAndVerify(source).VerifyIL("C..cctor()", @"
+{
+  // Code size        7 (0x7)
+  .maxstack  1
+  IL_0000:  ldc.i4.1
+  IL_0001:  stsfld     ""int C.x""
+  IL_0006:  ret
+}");
+        }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_09()
+        {
+            string source = @"
+#nullable enable
+
+struct S
+{
+    public int x;
+}
+
+class C
+{
+    static S? s1 = null;
+}";
+            CompileAndVerify(
+                source,
+                symbolValidator: validator,
+                options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All));
+
+            void validator(ModuleSymbol module)
+            {
+                var type = module.ContainingAssembly.GetTypeByMetadataName("C");
+                Assert.Null(type.GetMember(".cctor"));
+            }
+        }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_10()
+        {
+            string source = @"
+#nullable enable
+
+struct S
+{
+    public int x;
+}
+
+class C
+{
+    static S? s1 = default;
+    static S? s2 = default(S?);
+    static S? s3 = null;
+    static S? s4 = new S?();
+}";
+            // note: we could make the synthesized constructor smarter and realize that
```

Comment:
>note: we could make the synthesized constructor smarter [](start = 15, length = 55)

Which of the above cases causes the .cctor to be emitted? (Based on the previous test, it looks like `= null` does not result in a .cctor.) Consider splitting the test into two, with just the case(s) that result in .cctor in one and the rest in the other.